### PR TITLE
Adds support for graphite tags

### LINKED ISSUE #161: Flaky test
```
--- FAIL: TestSendMetricsAsync (0.01s)
	Error Trace:	graphite_test.go:125
			asm_amd64.s:2337
	Error:      	Received unexpected error:
	            	accept tcp 127.0.0.1:32810: use of closed network connection
```
https://travis-ci.org/atlassian/gostatsd/builds/354706831#L874

### PR DESCRIPTION
**Pull request by tiedotguy**
This does a few things:
- Adds support for graphite tags (including a bunch of refactor work), however only using actual tags (#233, #213), and not node style (which is probably what was intended with #78).  I'm calling all 3 resolved by this, however.  If someone actually wants node style tags, a fresh issue should be opened.
- Pulls backend configuration to its own documentation file, because it's getting noisy.
- Fixed #161

### CODE REVIEW

File: pkg/backends/graphite/graphite_test.go [hunk 1]
```diff
@@ -135,7 +145,6 @@ func TestSendMetricsAsync(t *testing.T) {
 		}
 	}()
 	defer acceptWg.Wait()
-	defer l.Close()
```
Comment (tiedotguy) [inline pkg/backends/graphite/graphite_test.go]: This is #161, this is:
- duplicate a line from above
- executing before `acceptWg.Wait()`, leading to the socket being closed before the goroutine has accepted from it

Comment (ash2k) [reply pkg/backends/graphite/graphite_test.go]: I think the idea here was to close the socket and unblock the accept goroutine if something goes wrong in the main one. It should be ok to close multiple times, I think. Anyway, I'm not saying this is correct code and doesn't have bugs. Just trying to recollect why it is like that and I may be misremembering. Not opposed to making this change at all.

Comment (tiedotguy) [reply pkg/backends/graphite/graphite_test.go]: Makes sense.  I can re-factor it if we start seeing failures again for sure, but this was close enough.

File: README.md [hunk 2]
```diff
@@ -119,81 +119,18 @@ There is no capability to run an https server at this point in time, and no auth
...
-- Directly to the Insights Collector - [Insights Event API](https://docs.newrelic.com/docs/insights/insights-data-sources/custom-data/send-custom-events-event-api)
-- Via the Infrastructure Agent's inbuilt HTTP Server
-
-### [New Relic Insights Event API](https://docs.newrelic.com/docs/insights/insights-data-sources/custom-data/send-custom-events-event-api)
-Sending directly to the Event API alleviates the requirement of needing to have the New Relic Infrastructure Agent. Therefore you can run this from nearly anywhere for maximum flexibility. This also becomes a shorter data path with less resource requirements becoming a simpler setup.
-
-To use this method, create an Insert API Key from here: https://insights.newrelic.com/accounts/YOUR_ACCOUNT_ID/manage/api_keys
+Configuring backends
+--------------------
+Refer to [BACKENDS.md] for configuration options for the backends
```
Comment (juliawong) [inline README.md]: Is this link meant to be clickable? Because when previewing it I see the text `[BACKENDS.md]`

Comment (tiedotguy) [reply README.md]: I'm not too sure, actually.  It's the style used throughout the file when referencing local files, so I just followed it.

**Pull request merged**